server/src/main/java/org/elasticsearch/index/query/IntervalsSourceProvider.java

```diff
@@ -746,6 +752,124 @@ String getUseField() {
         }
     }
 
+    public static class Regexp extends IntervalsSourceProvider {
+
+        public static final String NAME = "regexp";
+
+        private final String pattern;
+        private final String analyzer;
+        private final String useField;
+
+        public Regexp(String pattern, String analyzer, String useField) {
+            this.pattern = pattern;
+            this.analyzer = analyzer;
+            this.useField = useField;
+        }
+
+        public Regexp(StreamInput in) throws IOException {
+            this.pattern = in.readString();
+            this.analyzer = in.readOptionalString();
+            this.useField = in.readOptionalString();
+        }
+
+        @Override
+        public IntervalsSource getSource(SearchExecutionContext context, MappedFieldType fieldType) {
+            NamedAnalyzer analyzer = null;
+            if (this.analyzer != null) {
+                analyzer = context.getIndexAnalyzers().get(this.analyzer);
+            }
+            if (useField != null) {
+                fieldType = context.getFieldType(useField);
+                assert fieldType != null;
+            }
+            if (analyzer == null) {
+                analyzer = fieldType.getTextSearchInfo().searchAnalyzer();
+            }
+            BytesRef normalizedPattern = analyzer.normalize(fieldType.name(), pattern);
+            IntervalsSource source = fieldType.regexpIntervals(normalizedPattern, context);
+            if (useField != null) {
+                source = Intervals.fixField(useField, source);
+            }
+            return source;
+        }
+
+        @Override
+        public void extractFields(Set<String> fields) {
+            if (useField != null) {
+                fields.add(useField);
+            }
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Regexp regexp = (Regexp) o;
+            return Objects.equals(pattern, regexp.pattern)
+                && Objects.equals(analyzer, regexp.analyzer)
+                && Objects.equals(useField, regexp.useField);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(pattern, analyzer, useField);
+        }
+
+        @Override
+        public String getWriteableName() {
+            return NAME;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
```

Comment:
@mayya-sharipova while looking over the diff between the `lucene_snapshot` branch and upstream, I noticed this new named writable. Will this need to be accompanied by a new transport version - to indicate the change/addition of the new serialisable type?